### ISSUE TITLE
Using factory with Mongo.ObjectID collections

### ISSUE THREAD

**Issue by glennnaessens** [versolearning/meteor-factory#25]
My collections use Mongo.ObjectID's for _id generation.
`export const Clients = new Mongo.Collection('clients', { idGeneration: 'MONGO' });`

However, factory seems to default to a `Random.Id()` id generation which fails a lot of tests for me, so I tried adding id generation right into factory definition:
`Factory.define('client', Clients, {
	_id: new Mongo.ObjectID(),
//...
});`

This results in a mongo duplicate id error when I call `Factory.create('client')` three times in a row, though. For some reason all three clients end up with the same `_id` field. 

Please advise.

**Comment by guylemon** [comment]
Hey @glennnaessens when you `define` the factory, the attributes you pass in will be static.  To get the object id to be unique, try doing it this way-

```
Factory.create('client', { _id: new Mongo.ObjectID() })
```

**Comment by dburles** [comment]
@guylemon that's right, you can also pass a function so that it's generated at execution time. e.g:
```js
Factory.define('client', Clients, { _id: () => new Mongo.ObjectID(), //... });
```

**Comment by guylemon** [comment]
@dburles Nice!  That is even better. Thanks for the effort and time you put into creating this tool!

**Comment by dburles** [comment]
@guylemon thanks!

**Comment by glennnaessens** [comment]
I'm sorry about the late reply, project got put on ice for a wee bit. 
Thank you both for your reply. I got it working using @guylemon's suggestion. When I used @dburles' method however, I got an error stating Meteor only accepts Non-empty strings or Mongo ObjectID's as _id. Is it possible the _id function isn't executed but passed as a whole into the database?

**Comment by glennnaessens** [comment]
Actually it does run on execution as `{ just_an_id: () => new Mongo.ObjectID(), ...}` gets its id no problem but it's not happy about using that technique on the `_id` field for some reason.

It seems I will need something that runs on execution anyway because using the `Factory.get('company')` for relational stuff still throws a company with a String _id at me.

**Comment by dburles** [comment]
@glennnaessens Yeah, you're right. The package is pretty tied into using Meteor's `Random.id()`, however, support for ObjectID's would be a nice addition. I'd be more than happy to accept a PR.

**Comment by glennnaessens** [comment]
Ask and ye shall receive https://github.com/versolearning/meteor-factory/pull/28
